test_runner/regress/test_ondemand_download.py

```diff
@@ -656,5 +658,225 @@ def test_compaction_downloads_on_demand_with_image_creation(neon_env_builder: Ne
     assert dict(kinds_after) == {"Delta": 4, "Image": 1}
 
 
+def test_ondemand_download_cancelled_by_config_location(neon_env_builder: NeonEnvBuilder):
```

Comment:
Yeah good nag, will add those.

Reply:
Added a hint of shutdown, but I was never hunting, more like demonstrating. I just didn't want to do this test originally because I was thinking it cannot be done without being very flaky, and the unit tests existed already.